pyOCD/tools/pyocd.py

```diff
@@ -218,8 +218,13 @@ def run(self):
                 else:
                     dumpHexData(data, args.read, width=args.width)
             elif args.action == ACTION_WRITE:
+                is_flash = self.isFlashWrite(args)
                 if args.width == 8:
-                    target.writeBlockMemoryUnaligned8(args.write, args.data)
+                    if is_flash:
+                        target.flash.init()
+                        target.flash.programPhrase(args.write, args.data)
```

Comment:
I've taken notes on these ideas. And implemented the minimum program length.

Unfortunately, you can't write individual bytes, at least on Kinetis. If you try to write to a phrase that has been written previously, even if you are only writing to bytes that are 0xFF and previously programmed bytes in that phrase are set to the same value, the flash gets unhappy.